dds_web/api/project.py

```diff
@@ -115,38 +116,56 @@ def post(self):
             # Can only be Deleted if never made Available
             if project.has_been_available:
                 raise DDSArgumentError(
-                    "Project cannot be deleted if it has ever been made available, instead Abort it from Available"
+                    "Project cannot be deleted if it has ever been made available, abort it instead"
                 )
+            project.is_active = False
 
         # Moving to Archived
         if new_status == "Archived":
             is_aborted = extra_args.get("is_aborted", False)
+            if project.current_status == "In Progress":
+                if not (project.has_been_available and is_aborted):
+                    raise DDSArgumentError(
+                        "Project cannot be archived from this status but can be aborted if it has ever been made available"
+                    )
+            project.is_active = False
 
         add_status = models.ProjectStatuses(
             **{"project_id": project.id, "status": new_status, "date_created": curr_date},
             deadline=add_deadline,
             is_aborted=is_aborted,
         )
+        delete_message = ""
         try:
             project.project_statuses.append(add_status)
-            db.session.commit()
-            if new_status in ["Deleted", "Archived"]:
-                # TODO Call function to delete files
-                print("TODO", flush=True)
+            if not project.is_active:
+                # Deletes files (also commits session in the function - should it be decoupled?)
```

Comment:
We can keep it like it is for now and possibly refactor later 👍🏻

Reply:
👍